### PR TITLE
Introduce metricslib

### PR DESCRIPTION
**Pull request by vctrfrnndz**
Intruduce new metrics lib 

For review @cristiandouce

### CODE REVIEW

File: _layouts/default.html [hunk 1]
```diff
@@ -138,8 +198,8 @@
         });
 
         window.login = function login() {
-          if (window.analytics) {
-            window.analytics.track('click:signin', getBasicInfo());
+          if (metricsLib) {
```
Comment (mgonto) [inline _layouts/default.html]: Is the if needed? /cc @cristiandouce it should work anyway, right?

File: _layouts/default.html [hunk 2]
```diff
@@ -286,6 +348,8 @@
               initialize();
 
               NProgress.done();
+
+              metricsLib.page();
```
Comment (mgonto) [inline _layouts/default.html]: Does this run on first load as well?

Comment (vctrfrnndz) [inline _layouts/default.html]: Yes. Also just updated PR with some fixes. Should be good to go with @cristiandouce 's blessing

File: _layouts/default.html [hunk 3]
```diff
@@ -121,13 +172,13 @@
         };
 
         lock.on('signup ready', function () {
-          if (!window.analytics) return;
-          window.analytics.track('open:signup', getBasicInfo());
+          if (!metricsLib) return;
```
Comment (cristiandouce) [inline _layouts/default.html]: always use `window.metricsLib` please

Comment (cristiandouce) [inline _layouts/default.html]: or define a getter function like
```js
function getMetrics() {
  return window.metricsLib;
}
// later

var metrics = getMetrics();

if (!metrics) { return; };
metrics.something()
```

**Comment by gnandretta** [comment]
@cristiandouce I've verified the following events show up in Segment's debugger:
- click:signin
- open:signin
- click:signup
- open:signup
- success:blog-signup
- pages (when visiting for the first time and changes via pjax)

**Comment by mgonto** [comment]
Cool :). Thanks @gnandretta and good catch. Ready to merge @cristiandouce right?

**Comment by cristiandouce** [comment]
All good! :+1:

**Pull request merged**